universal_transfer_operator/src/universal_transfer_operator/data_providers/filesystem/sftp.py

```diff
@@ -114,7 +116,14 @@ def get_complete_url(self, dst_url: str, src_url: str) -> str:
 
         return urlunparse(final_url)
 
-    def write_using_smart_open(self, source_ref: FileStream) -> str:
+    def write_using_smart_open(self, source_ref: FileStream | pd.DataFrame):
+        """Write the source data from remote object i/o buffer to the dataset using smart open"""
+        if isinstance(source_ref, FileStream):
+            return self.write_file_using_smart_open(source_ref=source_ref)
+        elif isinstance(source_ref, pd.DataFrame):
+            return self.write_dataframe_using_smart_open(source_ref=source_ref)
```

Comment:
Don't we want to return `source_ref` in this case?

Reply:
I think we should return `self.dataset.path` which is the destination value. source_ref will contain the value where data is being read from. It can be local worker path as well in some cases, which is not useful to end user. 

Also for being consistent with existing code we are giving the value of destination, which user already have, maybe we can provide something like `bytes transferred` or `rows affected` instead? but this has to be a separate ticket, WDYT?